### ISSUE TITLE
should we issue errors for options that do not affect soundness? 

### ISSUE THREAD

**Issue by jmesserly** [dart-lang/sdk#30422]
Analyzer/DDC respect all strong mode options, including `no-implicit-casts` and `no-implicit-dynamic`, whether supplied via the command line or analysis_options file.

Currently both of those options result in new errors flagged in the code. This causes confusion from users, as neither option affects soundness of strong mode, and our strong mode implementations (currently: DDC) can happily compile code whether those options are on are off (indeed, neither option is used by the compiler).

Perhaps they should be warnings when those flags are turned on, so they are printed, but it is clear that they do not prevent compiling or running code.

Thoughts? @leafpetersen

**Comment by jmesserly** [comment]
for context see this thread: https://github.com/dart-lang/sdk/issues/29713. The discussion was framed in terms of DDC, but I suspect we went wrong in choosing the error level. In general, DDC absolutely does need to respect any strong mode/type system options (e.g. when we experimented with non-nullability or other type system changes). But these particular options feel problematic, because they are purely informative messages.

**Comment by jmesserly** [comment]
another thought: I think we (or at least, me) got in a mindset of "all warnings are errors" because so many of Dart 1's "static warnings" were precisely the strong mode soundness things that we need to be errors. But, warnings can still be a very useful message level :). "hey this is bad, but it doesn't prevent us compiling"

**Comment by bwilkerson** [comment]
I agree, but I think this is more of a language issue. We need to decide whether the spec will include only errors or include both errors and warnings. We also need to decide which warnings will be supported across tools and which will be specific to the analyzer.

**Comment by leafpetersen** [comment]
Pending resolution of larger questions, I'm happy to have these moved to warnings if it unblocks things.
